### ISSUE TITLE
(aws_cdk.aws_cloudfront): "edge_lambdas" parameters error in "default_behavior(BehaviorOptions)" of "aws_cdk.aws_cloudfront.Distribution"

### ISSUE THREAD

**Issue by terrificdm** [aws/aws-cdk#16068]
### Reproduction Steps

When I put "edge_lambdas" parameters in "default_behavior"of "aws_cdk.aws_cloudfront.Distribution", and make "cdk synth", it always pops up with an error like blow descriptions:

```sh
jsii.errors.JavaScriptError: 
  Error: Missing required properties for @aws-cdk/aws-cloudfront.EdgeLambda: eventType, functionVersion
      at validateRequiredProps (/tmp/tmpxbbq5tx2/lib/program.js:9395:23)
      at Object.deserialize (/tmp/tmpxbbq5tx2/lib/program.js:9065:29)
      at Kernel._toSandbox (/tmp/tmpxbbq5tx2/lib/program.js:8505:69)
      at value.map.x (/tmp/tmpxbbq5tx2/lib/program.js:8974:49)
      at Array.map (<anonymous>)
      at Object.deserialize (/tmp/tmpxbbq5tx2/lib/program.js:8974:34)
      at Kernel._toSandbox (/tmp/tmpxbbq5tx2/lib/program.js:8505:69)
      at mapValues (/tmp/tmpxbbq5tx2/lib/program.js:9070:37)
      at mapValues (/tmp/tmpxbbq5tx2/lib/program.js:9341:35)
      at Object.deserialize (/tmp/tmpxbbq5tx2/lib/program.js:9066:28)
```

And my codes are as below:

```python
from aws_cdk import (
    core,
    aws_s3 as s3,
    aws_s3_deployment as s3deploy,
    aws_ec2 as ec2,
    aws_iam as iam,
    aws_cloudfront as cloudfront,
    aws_cloudfront_origins as origins,
    aws_lambda as lambda_
    )


class CloudfrontCdkStack(core.Stack):

    def __init__(self, scope: core.Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        # Create a S3 bucket to hold flask static content as well as flask program for EC2 to download and run
        assets_bucket = s3.Bucket(self, "AssetsBucket")
        
        s3deploy.BucketDeployment(self, "StaticAssets",
            sources=[s3deploy.Source.asset("./flask-demo")],
            destination_bucket=assets_bucket,
            )
                                  
        # Create an EC2 to run flask program which generates the dynamic content       
        vpc = ec2.Vpc.from_lookup(self, "VPC", is_default=True)
        
        amzn_linux = ec2.MachineImage.latest_amazon_linux(
            generation=ec2.AmazonLinuxGeneration.AMAZON_LINUX_2
            )
            
        iam_role = iam.Role(self, "InstanceReadS3", assumed_by=iam.ServicePrincipal("ec2.amazonaws.com"))
        iam_role.add_managed_policy(iam.ManagedPolicy.from_aws_managed_policy_name("AmazonS3ReadOnlyAccess"))
        
        instance = ec2.Instance(self, "Instance",
            instance_type=ec2.InstanceType("t3.micro"),
            machine_image=amzn_linux,
            vpc=vpc,
            key_name="demo",# You need to modify the key_name with your key-pairs name!
            role=iam_role
            )
            
        instance.user_data.add_s3_download_command(
            bucket=assets_bucket,
            bucket_key="flask.zip"
            )
            
        instance.user_data.add_s3_download_command(
            bucket=assets_bucket,
            bucket_key="start.sh"
            )
            
        instance.user_data.add_commands("cd /tmp && chmod +x start.sh && ./start.sh")
        
        instance.connections.allow_from_any_ipv4(
            ec2.Port.tcp(22), description="Allow ssh from internet")
        instance.connections.allow_from_any_ipv4(
            ec2.Port.tcp(80), description="Allow http from internet")
        
        # Associate a Lambda@Edge function with a specific behavior of CloudFront distribution    
        lambdafunc = lambda_.Function(self, "LambdaFunction",
            runtime=lambda_.Runtime.PYTHON_3_8,
            handler="lambda_function.lambda_handler",code=lambda_.Code.from_asset("./functions/lambda")
            )
            
        distribution = cloudfront.Distribution(self, "MyDist",
            default_behavior={
                "origin": origins.HttpOrigin(instance.instance_public_dns_name,
                    protocol_policy=cloudfront.OriginProtocolPolicy.HTTP_ONLY
                    ),
                "viewer_protocol_policy": cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
                "cache_policy": cloudfront.CachePolicy.CACHING_DISABLED,
                "edge_lambdas": [{
                    "function_version": lambdafunc.current_version,
                    "event_type": cloudfront.LambdaEdgeEventType.ORIGIN_REQUEST
                    }]
                },
            additional_behaviors={
                "/static/*": {
                    "origin": origins.S3Origin(assets_bucket),
                    "viewer_protocol_policy": cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
                    "cache_policy": cloudfront.CachePolicy.CACHING_OPTIMIZED,
                    # "edge_lambdas": [{
                        
                    #     "function_version": lambdafunc.current_version,
                    #     "event_type": cloudfront.LambdaEdgeEventType.ORIGIN_REQUEST
                    #     }]
                    }
                }
            )
        
```
And if I comment out the "edge_lambdas" in "default_behavior" and put those code in the  "additional_behaviors", then it works... Given the codes are same, I think it should be a bug and need to be fixed. 

code are as below:

```python
from aws_cdk import (
    core,
    aws_s3 as s3,
    aws_s3_deployment as s3deploy,
    aws_ec2 as ec2,
    aws_iam as iam,
    aws_cloudfront as cloudfront,
    aws_cloudfront_origins as origins,
    aws_lambda as lambda_
    )


class CloudfrontCdkStack(core.Stack):

    def __init__(self, scope: core.Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        # Create a S3 bucket to hold flask static content as well as flask program for EC2 to download and run
        assets_bucket = s3.Bucket(self, "AssetsBucket")
        
        s3deploy.BucketDeployment(self, "StaticAssets",
            sources=[s3deploy.Source.asset("./flask-demo")],
            destination_bucket=assets_bucket,
            )
                                  
        # Create an EC2 to run flask program which generates the dynamic content       
        vpc = ec2.Vpc.from_lookup(self, "VPC", is_default=True)
        
        amzn_linux = ec2.MachineImage.latest_amazon_linux(
            generation=ec2.AmazonLinuxGeneration.AMAZON_LINUX_2
            )
            
        iam_role = iam.Role(self, "InstanceReadS3", assumed_by=iam.ServicePrincipal("ec2.amazonaws.com"))
        iam_role.add_managed_policy(iam.ManagedPolicy.from_aws_managed_policy_name("AmazonS3ReadOnlyAccess"))
        
        instance = ec2.Instance(self, "Instance",
            instance_type=ec2.InstanceType("t3.micro"),
            machine_image=amzn_linux,
            vpc=vpc,
            key_name="demo",# You need to modify the key_name with your key-pairs name!
            role=iam_role
            )
            
        instance.user_data.add_s3_download_command(
            bucket=assets_bucket,
            bucket_key="flask.zip"
            )
            
        instance.user_data.add_s3_download_command(
            bucket=assets_bucket,
            bucket_key="start.sh"
            )
            
        instance.user_data.add_commands("cd /tmp && chmod +x start.sh && ./start.sh")
        
        instance.connections.allow_from_any_ipv4(
            ec2.Port.tcp(22), description="Allow ssh from internet")
        instance.connections.allow_from_any_ipv4(
            ec2.Port.tcp(80), description="Allow http from internet")
        
        # Associate a Lambda@Edge function with a specific behavior of CloudFront distribution    
        lambdafunc = lambda_.Function(self, "LambdaFunction",
            runtime=lambda_.Runtime.PYTHON_3_8,
            handler="lambda_function.lambda_handler",code=lambda_.Code.from_asset("./functions/lambda")
            )
            
        distribution = cloudfront.Distribution(self, "MyDist",
            default_behavior={
                "origin": origins.HttpOrigin(instance.instance_public_dns_name,
                    protocol_policy=cloudfront.OriginProtocolPolicy.HTTP_ONLY
                    ),
                "viewer_protocol_policy": cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
                "cache_policy": cloudfront.CachePolicy.CACHING_DISABLED,
                # "edge_lambdas": [{
                #     "function_version": lambdafunc.current_version,
                #     "event_type": cloudfront.LambdaEdgeEventType.ORIGIN_REQUEST
                #     }]
                },
            additional_behaviors={
                "/static/*": {
                    "origin": origins.S3Origin(assets_bucket),
                    "viewer_protocol_policy": cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
                    "cache_policy": cloudfront.CachePolicy.CACHING_OPTIMIZED,
                    "edge_lambdas": [{
                        
                        "function_version": lambdafunc.current_version,
                        "event_type": cloudfront.LambdaEdgeEventType.ORIGIN_REQUEST
                        }]
                    }
                }
            )

```

### What did you expect to happen?

Successfully associate a L@E function with a CF's  default behavior.

### What actually happened?

Got an error as above description.

### Environment

  - **CDK CLI Version  :*1.117.0*
  - **Framework Version:*1.118.0*
  - **Node.js Version:*v10.24.1* 
  - **OS               :*Amaz Linux 2*
  - **Language (Version):*python 3.7.10* 

### Other

--- 

This is :bug: Bug Report

**Comment by ryparker** [comment]
Hey @terrificdm 👋🏻 

Thanks for bringing this to our attention.

A dev will pick up this bug in order of priority. This is marked as p1 which means it has been prioritized as important, although please keep in mind that we do have a large number of issues at the moment. It may be some time before we are able to solve this particular issue. We use +1s to help us prioritize our work, and as always we are happy to take contributions if anyone is interested to pick this up and submit a PR.

**Comment by terrificdm** [comment]
Just want to add some comments related to putting "edge_lambdas" into "additional_behaviors" of distribution. It turns out: even though the "cdk synth" command can be conducted without any error notification, the parameters which were added or modified for "additional_behaviors" didn't take effect. Seems "every parameters" of "additional_behaviors" are always with default settings no matter how you change those.

**Comment by terrificdm** [comment]
Just want to add some comments related to putting "edge_lambdas" into "additional_behaviors" of distribution. It turns out: even though the "cdk synth" command can be conducted without any error notification, the parameters which were added or modified for "additional_behaviors" didn't take effect. Seems "every parameters" of "additional_behaviors" are always with default settings no matter how you change those.
BTW, I have changed all my python codes into typescript, and everything goes well.
